### PR TITLE
Namespace has no effect when exposing deployment with --dry-run=client

### PR DESCRIPTION
**Pull request by masap**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
This is how to reproduce.

```
$ kubectl create ns testns
$ kubectl -n testns create deployment test-deploy --image=nginx
$ kubectl -n testns expose deployment.apps/test-deploy --port=80 --dry-run=client -o yaml
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    app: test-deploy
  name: test-deploy
spec:
  ports:
  - port: 80
    protocol: TCP
    targetPort: 80
  selector:
    app: test-deploy
status:
  loadBalancer: {}
```

If ```--dry-run=client``` is not specified, namespace element is contained in the yaml.

**Which issue(s) this PR fixes**:
NONE

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
Fixed no effect namespace when exposing deployment with --dry-run=client.
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go
Comment (knight42) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go:630]: why is this the expected output?

**Review by knight42 — COMMENTED** [review]
Nice catch!

/ok-to-test

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go
Comment (masap) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go:630]: Though exact result is as follows, this code checks only namespace because we are only interested in it.

```
apiVersion: v1
kind: Service
metadata:
  creationTimestamp: null
  labels:
    svc: test
  name: foo
  namespace: testns
spec:
  ports:
  - port: 14
    protocol: UDP
    targetPort: 14
  selector:
    func: stream
  type: LoadBalancer
status:
  loadBalancer: {}
```

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (cblecker) [inline staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:340]: @soltysh @deads2k Is it preferable to take this route, and manipulate the generated object before printing.. or would it be better to, [in the generator](https://github.com/kubernetes/kubectl/blob/master/pkg/generate/versioned/service.go#L80), add the namespace to the generated object?

Comment (masap) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:340]: That code seems to be used in the ```run``` command as well.

https://github.com/kubernetes/kubernetes/blob/8b24d1f2f92063faa4cff1cdea6a3d504597b555/staging/src/k8s.io/kubectl/pkg/cmd/run/run.go#L677-L679

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go
Comment (masap) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose_test.go:630]: @knight42 Is there any comment ?

**Comment by knight42** [comment]
/lgtm
Need approval from @kubernetes/sig-cli-maintainers.

**Comment by masap** [comment]
@knight42 Thanks!

File: staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go
Comment (soltysh) [reply staging/src/k8s.io/kubectl/pkg/cmd/expose/expose.go:340]: This is ok temporary solution, we're slowly getting rid of generators which will clear this situation.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve

**Pull request merged**
